client/src/annoMatrix/middleware.ts

```diff
@@ -35,39 +42,39 @@ function _itIsTimeForGC() {
   return false;
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-function _doGC(store: any) {
+function _doGC(store: MiddlewareAPI): void {
   const state = store.getState();
 
   // these should probably be a function imported from undoable.js, etc, as
-  // they have overly intimiate knowledge of our reducers.
+  // they have overly intimate knowledge of our reducers.
   const undoablePast = state["@@undoable/past"];
   const undoableFuture = state["@@undoable/future"];
   const undoableStack = undoablePast
     .concat(undoableFuture)
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- TODO revisit: waiting for typings from /reducers
     .flatMap((snapshot: any) =>
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      snapshot.filter((v: any) => v[0] === "annoMatrix").map((v: any) => v[1])
+      snapshot
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any --- TODO revisit: waiting for typings from /reducers
+        .filter((v: any) => v[0] === "annoMatrix")
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any --- TODO revisit: waiting for typings from /reducers
+        .map((v: any) => v[1])
     );
   const currentAnnoMatrix = state.annoMatrix;
 
   /*
   We want to identify those matrixes currently "hot", ie, linked from the current annoMatrix,
   as our current gc algo is more aggressive with those not hot.
   */
-  const allAnnoMatrices = new Map(
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    undoableStack.map((m: any) => [m, { isHot: false }])
+  const allAnnoMatrices = new Map<AnnoMatrix, GCHints>(
+    undoableStack.map((m: AnnoMatrix) => [m, { isHot: false }])
   );
   let am = currentAnnoMatrix;
   while (am) {
```

Comment:
this loop needs to be updated to reflect the new viewOf protocol (ie, it isn't null on the base).   I would just change the while predicate to be something like `while (am && am.isView) { ... }`

Reply:
Fixed defect in [middleware:L72](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/middleware.ts#L72).